nibabel/tests/test_processing.py

```diff
@@ -426,3 +427,33 @@ def test_against_spm_resample():
     moved2output = resample_to_output(moved_anat, 4, order=1, cval=np.nan)
     spm2output = nib.load(pjoin(DATA_DIR, 'reoriented_anat_moved.nii'))
     assert_spm_resampling_close(moved_anat, moved2output, spm2output);
+
+
+def test__transform_range():
+    assert_array_equal(_transform_range([2, 4, 6], -1, 1), [-1, 0, 1])
+    assert_array_equal(_transform_range([-1, 0, 1], 2, 6), [2, 4, 6])
+    assert_array_equal(_transform_range(np.arange(11), 0, 5),
+                       np.arange(0, 5.5, 0.5))
+    assert_array_equal(_transform_range(np.arange(-100, 101), 0, 200),
+                       np.arange(201))
+
+
+@needs_scipy
+def test_conform():
+    anat = nib.load(pjoin(DATA_DIR, 'anatomical.nii'))
+
+    c = conform(anat)
+    assert c.shape == (256, 256, 256)
+    assert c.header.get_zooms() == (1, 1, 1)
+    assert c.dataobj.dtype == np.dtype(np.uint8)
+    assert aff2axcodes(c.affine) == ('R', 'A', 'S')
+
+    c = conform(anat, out_shape=(100, 100, 200), voxel_size=(2, 2, 1.5))
+    assert c.shape == (100, 100, 200)
+    assert c.header.get_zooms() == (2, 2, 1.5)
+    assert c.dataobj.dtype == np.dtype(np.uint8)
+    assert aff2axcodes(c.affine) == ('R', 'A', 'S')
+
+    # Error on non-3D images.
```

Comment:
is that what freesurfer does?  I thought it might be nice to have this one applicable to 4D (or 5D - AFNI etc) series, to "conform" the first 3 dimensions to desired resolution/orientation etc

Reply:
i don't think it is what freesurfer does, but for simplicity and due to my lack of knowledge, i made it this way. how should the output shape and zooms be changed to support 4- and 5-D images?

Reply:
The output shape and zooms would not be affected by rolling the first three axes.

In any event, I'm okay with getting this in and adding 4/5D image support later, but if we're going to have a test, then we should mark it as one that we intend to relax. Otherwise it may look like a desired constraint to future developers.